examples/wrap/wrap_test.c

```diff
@@ -151,6 +151,15 @@ int TPM2_Wrapper_Test(void* userCtx)
     /* See if primary storage key already exists */
     rc = wolfTPM2_ReadPublicKey(&dev, &storageKey,
         TPM2_DEMO_STORAGE_KEY_HANDLE);
+
+#ifdef TEST_WRAP_DELETE_KEY
+    if (rc == 0) {
```

Comment:
Yes intentional. The `wolfTPM2_ReadPublicKey` will return 0 (TPM_RC_SUCCESS) if the key is found. So this will only attempt to delete the key if it exists. The statement below at 163 attempts to create the key if there is a failure.